src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdClientConfiguration.cs

```diff
@@ -97,6 +97,20 @@ public void Configure(string name, OpenIdConnectOptions options)
                     _logger.LogError("The client secret could not be decrypted. It may have been encrypted using a different key.");
                 }
             }
+
+            if (settings.Parameters.Any())
+            {
+                var parameters = settings.Parameters;
+                options.Events.OnRedirectToIdentityProvider = (context) =>
+                {
+                    foreach(var parameter in parameters)
+                    {
+                        context.ProtocolMessage.SetParameter(parameter.Name, parameter.Value);
```

Comment:
It's worth noting that `SetParameter()` allows overriding existing parameters. Do we want to support that kind of extensibility?

Reply:
I wondered if that would be a concern - but the extensibility is there anyway, this just reveals it via a ui, so adds more flexibility - something we need for a particular azure client connection

Reply:
I'm asking that because I regularly people trying to replace the default parameters already generated by an OIDC client without realizing it's often not a good idea. To emphasize that, you may want to rename the title to "Additional parameters" or add an "advanced" label somewhere 😃

Reply:
Also, I'm curious: why not using the Microsoft.Identity.Web integration for Azure?

Reply:
Updated to `Advanced Parameters` and I made a note in the docs as well. Hopefully suits @kevinchalet 😄 

Re : Microsoft.Identity.Web for azure.
I'm using that as well, just have (currently) two external providers, which both happen to be different variants of AzureAD, and only one of is after some extra params.

We're also tracking https://github.com/OrchardCMS/OrchardCore/issues/8752 upgrading the current AzureAD provider to `Microsoft.Identity.Web` as currently it's still on the older versions.